src/Raven.Server/ServerWide/Commands/Indexes/PutStaggeredIndexCommand.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.ServerWide;
+using Raven.Server.Rachis;
+using Raven.Server.Utils;
+using Sparrow;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.ServerWide.Commands.Indexes
+{
+    public class PutStaggeredIndexCommand : UpdateDatabaseCommand
+    {
+        public string IndexName { get; set; }
+
+        public PutStaggeredIndexCommand()
+        {
+            // for deserialization
+        }
+
+        public PutStaggeredIndexCommand(string databaseName, string indexName, string uniqueRequestId) 
+            : base(databaseName, uniqueRequestId)
+        {
+            IndexName = indexName;
+        }
+
+        public override string UpdateDatabaseRecord(DatabaseRecord record, long etag)
+        {
+            if (record.StaggeredIndexes.TryGetValue(IndexName, out var staggeredIndex ) == false)
+            {
+                return null; // was already removed
+            }
+
+            // What happens here though? No memebers.. do we resend the command? How do we make sure this will run again?
+            if (record.Topology.Members.Count == 0)
+                return null;
+
+            var reversedMembers = record.Topology.Members;
+            reversedMembers.Reverse();
+
+            // Check if we have new members, coming back from rehab or perhaps completely new
+            foreach (var member in reversedMembers)
+            {
+                if (staggeredIndex.ContainsKey(member) == false)
```

Comment:
Why bother with the check, and why does the order matter?
`TryAdd` will skip the addition if it is there, can just scan through this.